hts-sys/build.rs

```diff
@@ -43,114 +89,115 @@ fn main() {
-        .status().unwrap().success()
+    // write out version.h
     {
-        panic!("could not configure htslib nor any of its plugins")
+        let mut f = std::fs::File::create(out.join("htslib").join("version.h")).unwrap();
+        // FIXME - get version somehow.
+        writeln!(&mut f, "#define HTS_VERSION_TEXT \"1.0\"").unwrap();
     }
 
-    if !Command::new("make")
-        .current_dir(out.join("htslib"))
-        .arg(format!("CC={}", cc_path.display()))
-        .arg(format!("CFLAGS={}", &cc_cflags))
-        .arg("lib-static")
-        .status()
-        .unwrap()
-        .success()
+    cfg.file("wrapper.c");
+    cfg.compile("hts");
+
+    // If bindgen is enabled, use it
+    #[cfg(feature = "bindgen")]
     {
-        panic!("failed to build htslib");
+        bindgen::Builder::default()
+            .header("wrapper.h")
+            .generate_comments(false)
+            .blacklist_function("strtold")
+            .blacklist_type("max_align_t")
+            .generate()
+            .expect("Unable to generate bindings.")
+            .write_to_file(out.join("bindings.rs"))
+            .expect("Could not write bindings.");
     }
 
-    cfg.file("wrapper.c").compile("wrapper");
+    // If no bindgen, use pre-built bindings
+    #[cfg(all(not(feature = "bindgen"), target_os="macos"))]
+    {
+        fs::copy("osx_prebuilt_bindings.rs", out.join("bindings.rs"))
+            .expect("couldn't copy prebuilt bindings");
+        println!("cargo:rerun-if-changed=osx_prebuilt_bindings.rs");
+    }
 
-    bindgen::Builder::default()
-        .header("wrapper.h")
-        .generate_comments(false)
-        .blacklist_function("strtold")
-        .blacklist_type("max_align_t")
-        .generate()
-        .expect("Unable to generate bindings.")
-        .write_to_file(out.join("bindings.rs"))
-        .expect("Could not write bindings.");
+    #[cfg(all(not(feature = "bindgen"), target_os="linux"))]
+    {
+        fs::copy("linux_prebuilt_bindings.rs", out.join("bindings.rs"))
+            .expect("couldn't copy prebuilt bindings");
+        println!("cargo:rerun-if-changed=linux_prebuilt_bindings.rs");
+    }
 
     let include = out.join("include");
     fs::create_dir_all(&include).unwrap();
     if include.join("htslib").exists() {
         fs::remove_dir_all(include.join("htslib")).expect("remove exist include dir");
     }
     copy_directory(out.join("htslib").join("htslib"), &include).unwrap();
-    fs::copy(out.join("htslib").join("libhts.a"), out.join("libhts.a")).unwrap();
 
     println!("cargo:root={}", out.display());
     println!("cargo:include={}", include.display());
     println!("cargo:libdir={}", out.display());
-    println!("cargo:rustc-link-lib=static=hts"); // XXX: only for static, adapt for dynamic?
     println!("cargo:rerun-if-changed=wrapper.c");
     println!("cargo:rerun-if-changed=wrapper.h");
-    println!("cargo:rerun-if-changed=htslib/Makefile");
-    let globs = std::iter::empty()
```

Comment:
This essentially un-does PR https://github.com/rust-bio/rust-htslib/pull/204 from @nlhepler , but I guess it's fine with the new approach?

Reply:
hmm, in theory we should still rebuild if a header changes? I'll ask him about that.

Reply:
We will want to keep the rerun-if-changed directives for header files, as those are effectively dumped into the C files that include them by the preprocessor.

Reply:
@nlhepler restored.